codeforamerica/vita-min
# Nj 97 income eligibility

## Link to pivotal/JIRA issue

https://github.com/newjersey/affordability-pm/issues/97

## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main

## What was done?
- I moved logic that was going to be reused (wage calculation, age check, property tax eligibility) into `lib/efile/`

- I edited both `next_path()` and `show?()` for several of the property tax-related screens to skip questions not needed due to taxpayer not being eligible for what they were asking. (I learned that `next_path()` sort of overrides `show?()`, and had to use both to properly build the flow)

- I considered adding a line to the calculator (and a bunch of related tests) to make super explicit the requirement `40a, and 41 should be blank`. Those lines are blank if `property_tax_paid` or `rent_paid` are `nil`, which is always true when the screens this PR skips are skipped. So in the end it felt over-constrained and would not happen in practice so I removed the extra code. You can see what I removed in [this commit](https://github.com/codeforamerica/vita-min/commit/192b455313b4da2a69a37289aba44aed7c306c1a) if interested.

## How to test?
- The current flow stays the same for everyone with income above the minimum, so that can be still experienced with `Zeus many w2s` (video below)
- The `Minimal` transfer has an income below the minimum, so using that you can see that the user is still asked about being homeowner/tenant but then is moved along (video below)
- The `Blind primary` transfer has income below minimum but qualifies for credit-only, so you can see user is asked about homeowner/tenant and the housing details eligibility, but skips the screen asking for property tax/rent paid (video below)
- The new `Married filing jointly 15k wages` transfer has an income that is below the $20k minimum for MFJ
- The new `Married filing separately 15k wages` transfer has an income that is above the $10k minimum for MFS

## Videos

### Zeus many w2s



https://github.com/user-attachments/assets/bda9ab82-ccfe-43bc-80c5-b1143b387f70




---
### Minimal


https://github.com/user-attachments/assets/266c1a76-0060-43e4-923c-3254f29562d3



---
### Blind primary


https://github.com/user-attachments/assets/ab4a0f14-8c47-479e-807c-17eac2b60c5e

---
### MFJ $15k Wages (below $20k minimum)


https://github.com/user-attachments/assets/1291d313-bd8b-4557-8c64-e205027b1de1

---
### MFS $15k Wages (above $10k minimum)


https://github.com/user-attachments/assets/14a1275b-9d65-41e7-87a9-44f03d96c4f0
